crates/transaction-pool/src/metrics.rs

```diff
@@ -0,0 +1,36 @@
+//! Transaction pool metrics.
+
+use metrics::{describe_counter, register_counter, Counter};
+
+pub(crate) struct TxPoolMetrics {
```

Comment:
for metrics we want:

* it to be Optional
* shareabel

so I think Ideally this should look like

```rust
struct 
```TxPoolMetrics {
    inner: Arc<Option<TxPoolMetricsInner>>
}
```

where `TxPoolMetricsInner` is the type that has the counter

this way we could share it and use it in multiple pools for example

Reply:
Afaik the metrics are all shareable and "optional" already - if metrics are disabled by the consumer, these are effectively no-ops: https://docs.rs/metrics/latest/metrics/#in-libraries

Reply:
I see, so having Option would be an overhead if one is installed. and counters are already Option<Arc>

nvm then.

the metrics macros obfuscate a lot.

Reply:
They do, I do wonder if there is a more "hands on" API w/o macros